ratings/rate_subdir.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+sys.path.insert(0, '.')
+
+from absl import flags
+from absl import app
+
+import fnmatch
+import os
+import re
+from tqdm import tqdm
+from collections import defaultdict
+from ratings import math_ratings
+
+
+FLAGS = flags.FLAGS
+
+PW_REGEX = "PW\[([^]]*)\]"
+PB_REGEX = "PB\[([^]]*)\]"
+RESULT_REGEX = "RE\[([^]]*)\]"
+
+
+def extract_pairwise(files):
+    ids = defaultdict(lambda: len(ids))
```

Comment:
this is quite clever :)

Reply:
thank you :)  added a comment because i'll forget what it's for.